File: Retrofit Basic/app/src/main/java/com/derahh/retrofitbasic/MainActivity.java
package com.derahh.retrofitbasic;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.app.ProgressDialog;
import android.os.Bundle;

import com.derahh.retrofitbasic.adapter.CustomAdapter;
import com.derahh.retrofitbasic.model.PhotoData;
import com.derahh.retrofitbasic.services.ApiClient;
import com.derahh.retrofitbasic.services.GetService;

import java.util.List;

import retrofit2.Call;

public class MainActivity extends AppCompatActivity {

    private CustomAdapter adapter;
    private RecyclerView recyclerView;
    ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        progressDialog = new ProgressDialog(MainActivity.this);
        progressDialog.setMessage("Loading....");
        progressDialog.show();

        /*Create handle for the RetrofitInstance interface*/
        GetService service = ApiClient.getRetrofitInstance().create(GetService.class);
        Call<List<PhotoData>> call;
    }
}

File: RetrofitBasic/app/src/main/java/com/derahh/retrofitbasic/MainActivity.kt
package com.derahh.retrofitbasic

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.derahh.retrofitbasic.adapter.UserAdapter
import com.derahh.retrofitbasic.model.User
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val users = mutableListOf<User>()
        for(i in 0..100) {
            users.add(User("Dedi", "Rahmansah", "dedir2@gmail.com"))
        }

        recyclerView.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = UserAdapter(users)
        }
    }
}
